### ISSUE TITLE
Long delay in sending a reply or no reply at all, when the phone is locked.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#29]
```
What steps will reproduce the problem?
1. Reply to a sms within the client with the phone that has been locked due to 
inactivity.


What is the expected output? What do you see instead?
The expected output is that the client immediately (or at least within a 
minute) receives "Sending SMS to <contact>" notification in the chat window and 
the phone fires off the SMS to the contact. 

What I see instead is a delay. Sometimes more than a few minutes or so. When I 
receive an SMS, it has always been instantaneousness. My phone vibrates and I 
already have the SMS on my chat window. Whether the phone had its screen off or 
not, receiving SMS is instant.

However, when I send the SMS with a "cold" phone, a phone that is locked (not 
security lock) and the screen dark, it takes a few minutes at times or I give 
up and open up the phone to the homescreen. 

However, when I see that I'm not getting the "Sending sms to <contact>". I turn 
the screen on the phone and the SMS proceeds to send I receive the expected, 
SMS sent, SMS delivered. 


What version of the product are you using? On what operating system?
gtalksms: 1.6 (donate version). Phone: HTC Incredible 2.2. Desktop: Windows 7. 

Please provide any additional information below.
I've attached a screenshot that shows the timestamps. The first red circle 
shows how fast the Send SMS occured, while the 2nd circle did not send the SMS 
until I finally opened up the phone from its inactive state.

I can recreate this scenario pretty much every time.

```

Original issue reported on code.google.com by `i.vita...@gmail.com` on 5 Jan 2011 at 9:53

Attachments:
- [Capture.PNG](https://storage.googleapis.com/google-code-attachments/gtalksms/issue-29/comment-0/Capture.PNG)

**Comment by GoogleCodeExporter** [comment]
```
Sorry, my title is wonky. Should be: 

Long delay in sending a reply when phone is locked.
```

Original comment by `i.vita...@gmail.com` on 5 Jan 2011 at 9:57

**Comment by GoogleCodeExporter** [comment]
```
Issue 30 has been merged into this issue.
```

Original comment by `Florent....@gmail.com` on 5 Jan 2011 at 11:01

**Comment by GoogleCodeExporter** [comment]
```
I think it's due to the xmpp protocol latency and battery preserving on idle 
state.
```

Original comment by `Florent....@gmail.com` on 5 Jan 2011 at 11:05
- Changed title: **Long delay in sending a reply when phone is locked.**

**Comment by GoogleCodeExporter** [comment]
```
It seems that it only delays on sending a SMS/replying to a SMS. All other 
commands happen near instantly. Receiving SMS is also instant. 

Out of curiosity, is xmpp protocol absent from those functions?
```

Original comment by `i.vita...@gmail.com` on 6 Jan 2011 at 8:01

**Comment by GoogleCodeExporter** [comment]
```
Weird, tonight I've sent about 200 SMS (to test sms conversation new feature) 
and indeed about 100 SMS take between 5 and 10s to be sent, others are 
instantly sent.

With the new implementation of GTalkSMS we use severals intents to process a 
single message, it's maybe a lead to this issue...
```

Original comment by `Florent....@gmail.com` on 6 Jan 2011 at 11:26
- Changed state: **Investigating**

**Comment by GoogleCodeExporter** [comment]
```
FWIW, I too see gtalksms take quite some time to respond to commands when 
asleep - not quite a few minutes, but possibly up to 1 minute.  I've seen this 
forever - but can't reproduce it simply by pressing the power-button on the 
device.  I don't quite understand how sleep mode works on Android, but I 
suspect simply pressing that button so it *looks* asleep isn't quite the same 
as being truly asleep.

When I last saw this I grabbed the logs from the device and there is no 
evidence our process was killed - it just seemed slow to receive the request...
```

Original comment by `skippy.hammond@gmail.com` on 6 Jan 2011 at 11:44

**Comment by GoogleCodeExporter** [comment]
```
@florent - maybe add a log message to toe packet receiver in the xmppmanager - 
the logs might then help us work out if gtalksms is introducing the delay or 
the delay is before we are called.
```

Original comment by `skippy.hammond@gmail.com` on 7 Jan 2011 at 1:40

**Comment by GoogleCodeExporter** [comment]
```
For me, the long delay issue is only with the Reply command (even if the phone 
is awake). No issue with sms:contact:message method.
```

Original comment by `f.vers...@gmail.com` on 13 Jan 2011 at 9:07

**Comment by GoogleCodeExporter** [comment]
```
Issue 121 has been merged into this issue.
```

Original comment by `fschm...@gmail.com` on 20 Mar 2011 at 11:40

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
It seems like some Android Phones have different energy saving approaches. 
GTalkSMS > 2.0 and debug log enabled should allow us to track the arrival of a 
command and the sending of the resulting answer.
```

Original comment by `fschm...@gmail.com` on 20 Mar 2011 at 11:46
- Changed title: **Long delay in sending a reply or no reply at all, when the phone is locked.**

**Comment by GoogleCodeExporter** [comment]
```
I think it may be related to 
http://code.google.com/p/gtalksms/issues/detail?id=10
```

Original comment by `spouwny` on 4 May 2011 at 9:02

**Comment by GoogleCodeExporter** [comment]
```
@spouwny: GTalkSMS integrates nicely within Android. All services that get 
stopped because of low resources get restarted later (mostly within 10 seconds) 
by Android. After such a restart, GTalkSMS > 2.0 restores all previous 
configuration. So this two issues are very unlikely to be related.
```

Original comment by `fschm...@gmail.com` on 4 May 2011 at 11:16

**Comment by GoogleCodeExporter** [comment]
```
I have the same problem. I have a delay up to 5 minutes when the screen is 
locked and I use the reply command. 
```

Original comment by `emilorl...@gmail.com` on 20 Oct 2011 at 7:26

**Comment by GoogleCodeExporter** [comment]
```
@emilorloff and all others affected by this: Could you send us an app log ( 
http://code.google.com/p/gtalksms/wiki/GenerateAppLog ) when this happens?
```

Original comment by `fschm...@gmail.com` on 21 Oct 2011 at 8:06

**Comment by GoogleCodeExporter** [comment]
```
I have generated an app log and mailed it to you. 

The log documents that GTalkSMS receives the reply-command, and then it waits 
almost 2 minutes before doing anything.
```

Original comment by `emilorl...@gmail.com` on 22 Oct 2011 at 9:44

**Comment by GoogleCodeExporter** [comment]
```
GTalkSMS 3.3 makes use of wake-locks, please report if the problem is still 
there.
```

Original comment by `fschm...@gmail.com` on 16 Nov 2011 at 11:10
- Changed state: **UserFeedBack**

**Comment by GoogleCodeExporter** [comment]
```
I have been testing a bit today, and so far the delay has been gone. I will 
report back in a couple of days with a status. 
```

Original comment by `emilorl...@gmail.com` on 16 Nov 2011 at 6:23

**Comment by GoogleCodeExporter** [comment]
```
I have been testing for the last week and so and the problem seems to be fixed.
```

Original comment by `emilorl...@gmail.com` on 26 Nov 2011 at 4:13

**Comment by GoogleCodeExporter** [comment]
```
Fine, marking this as fixed.
```

Original comment by `fschm...@gmail.com` on 30 Nov 2011 at 10:09
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed and deployed with GTalkSMS 3.4
```

Original comment by `Florent....@gmail.com` on 7 Jan 2012 at 3:20
- Changed state: **Deployed**
